### ISSUE TITLE
Locker app itself is unsecured

### ISSUE THREAD

**Issue by fathermaurer** [Zyg0te/Locker#23]
This is an expansion on a concern in my Google Play Store review of Locker. 

Locker doesn't have protection against unauthorized changing of its own setting. Anyone who is given access to the phone when it is unlocked can change the settings of Locker. 

If possible, it would be ideal to have Locker require the owner's PIN/pattern/password before it will launch.

**Comment by Zyg0te** [comment]
Hmm, but whats the use case here. Locker limits the number of permitted unlock attempts which limits the ability to brute force the unlock code. If someone is given access to the phone when it's unlocked they have a number of ways to uninstall Locker completely. Any password protection I can implement in Locker will be worthless if someone is given access to your device in an unlocked state. 

Please enlighten me if I'm missing anything here.

**Comment by fathermaurer** [comment]
Hi there! Thanks for such a quick response - a perk of open source development that never ceases to delight me.

I'm likening access to Locker to having access to the Lock Screen settings in Android - which is only accessible by entering the PIN/pattern/password. This built-in protection prevents a user who has access to an unlocked device from locking out the rightful owner (say, by changing the PIN/pattern/password). Locker merits the same protection because it can trigger a wipe of the phone. 

A user uninstalling the app certainly reduces the security of the phone - but the danger I see is in the potential reconfiguration of Locker. There are a number of instances where a user might allow someone to use their unlocked device (children, co-worker, friend). Unlike the Lock Screen configuration, Locker configuration is immediately accessible. An example of a dangerous change would be to disable the warning and set Locker to wipe the device after one wrong attempt. The unwitting owner will get a nasty surprise next time they make a mistake in unlocking their device!

There are applications that can lock individual apps, but if a PIN/pattern/password challenge could be baked into Locker, it would help provide for it's own security.

**Comment by n76** [comment]
Assuming that you don't have a shortcut to Locker on your home screen, the person you hand the phone to would have to look for it. If it were me that you handed the phone to, and I was malicious, I'd be more likely to go to settings->apps to see what you have and I could do great damage there including uninstalling Locker which would break the lock failure erase.

My feeling is handing a person an unlocked phone is handing them the keys to the kingdom and trying to protect things after that is wasted effort.

**Comment by fathermaurer** [comment]
I can appreciate that perspective, but there a number of apps that have their own built-in protection to supplement the lock screen for just this kind circumstance. That protection allows for shared device use without opening access to everything (banking apps come to mind). Locker could use that same protection (as could its users!).

**Comment by Gaff** [comment]
I also subscribe to the keys-to-the-kingdom argument. If you give someone you don't trust completely access to your unlocked phone then all bets are off.

You can mitigate things a little with screen pinning, I use this whenever I let someone (usually a child) borrow my phone: http://gadgetguideonline.com/android/lollipop/how-to-use-screen-pinning-in-android-lollipop/

**Comment by fathermaurer** [comment]
We have migrated from a describing the app's vulnerabilities in potential use case to debating the use case itself. It's an interesting discussion, but maybe better suited to a forum or social network.

**Comment by kstefanini** [comment]
"better suited to a forum or social network"
Github is kinda both of these, and the best place to debate around key features, i think.

**Comment by fathermaurer** [comment]
Fair enough. I'm new to Github, so I'm still working out how it works here.

I'd prefer not to debate the issue further - having the feature baked into Locker allows for peace of mind for folks on either side of the issue, so if Zyg0te implements it, great! If not, Locker is still an extraordinarily useful app and I'm grateful for it.
